system_intelligence/hdd_info.py

```diff
@@ -1,108 +1,102 @@
+        if self.OS == 'linux':
+            import pyudev
+            self.context = pyudev.Context()
+        else:
+            pyudev = None
+            print('[bold yellow]Unable to import package pyudev. HDD information may be limited.')
+        self.HDD = pyudev is not None
+
+    def query_hdd(self):
+        """
+        Query info on any available HDDs on the users system
+        """
+        hdd_models = self.query_hdd_model()
+        hdd_usage = HddInfo.query_hdd_usage()
+
+        return {'model': hdd_models, 'usage': hdd_usage}
+
+    def query_hdd_model(self) -> t.Dict[str, dict]:
+        """
+        Get information about all hard drives.
+        """
+        if not self.HDD:
+            return {}
+        hdds = {}
+        for device in self.context.list_devices(subsystem='block', DEVTYPE='disk'):
+            if any(_ in device.device_path for _ in self.IGNORED_DEVICE_PATHS):
                 continue
-        usage = psutil.disk_usage(part.mountpoint)
-        hdd_to_usage[part.device] = {'total': bytes_to_hreadable_string(usage.total),
-                                     'used': bytes_to_hreadable_string(usage.used),
-                                     'free': bytes_to_hreadable_string(usage.free),
-                                     'percentage': str(usage.percent),
-                                     'fstype': part.fstype,
-                                     'mountpoint': part.mountpoint}
-
-    return hdd_to_usage
-
-
-def print_hdd_info(hdd_info: dict) -> None:
-    # Models
-    table = create_styled_table('Hard Disks Drives')
-
-    table.add_column('Disk Name', justify='left')
-    table.add_column('Model', justify='left')
-
-    for hdd, details in hdd_info['model'].items():
-        table.add_row(hdd, details['model'])
-        #  bytes_to_hreadable_string(details['size']) <- Removed since it may be misleading
-
-    console = Console()
-    console.print(table)
-
-    # Usage
-    table = create_styled_table('Disk Usage')
-
-    table.add_column('Device', justify='left')
-    table.add_column('Total', justify='left')
-    table.add_column('Used', justify='left')
-    table.add_column('Free', justify='left')
-    table.add_column('Use %', justify='left')
-    table.add_column('Type', justify='left')
-    table.add_column('Mount', justify='left')
-
-    for device, usage in hdd_info['usage'].items():
-        table.add_row(device,
-                      usage['total'],
-                      usage['used'],
-                      usage['free'],
-                      usage['percentage'],
-                      usage['fstype'],
-                      usage['mountpoint'])
-
-    console.print(table)
+            hdd = {'size': device.attributes.asint('size')}
+            for device_ in itertools.chain([device], device.ancestors):
+                try:
+                    hdd['model'] = device_.attributes.asstring('model')
+                    break
```

Comment:
what is this break for?

Reply:
Did not change the code here, so also have to take a look at the code